src/module/order/application/service/order.service.ts

```diff
@@ -6,16 +6,35 @@ import {
 import { Order, OrderStatus } from '../../domain/order.entity';
 import { UserService } from '../../../../module/user/application/service/user.service';
 import { RoleEnum } from '../../../../module/user/domain/user.entity';
-import { CannotUpdateOrderException } from '../errors/CannotUpdateOrder';
+import { CannotAccessOrderException, CannotUpdateOrderException } from '../errors/CannotUpdateOrder';
+import { Item } from 'src/module/item/domain/item.entity';
+import { ProductService } from 'src/module/product/application/service/product.service';
+import { ItemService } from 'src/module/item/application/service/item.service';
+import { DeleteResult } from 'typeorm';
 @Injectable()
 export class OrderService {
   constructor(
     @Inject(ORDER_REPOSITORY) private orderRepository: IOrderRepository,
     @Inject(UserService) private userService: UserService,
+    @Inject(ProductService) private productService: ProductService,
+    @Inject(ItemService) private itemService: ItemService,
   ) {}
 
-  async delete(userId: number, orderId: number): Promise<void> {
-    await this.findById(userId, orderId);
+  async delete(userId: number, orderId: number): Promise<DeleteResult> {
+    const user = await this.userService.findUserById(userId);
+    if (user.role === RoleEnum.admin) {
+      const orderToDelete = await this.findById(userId, orderId)
+      if (!orderToDelete) {
+        throw new NotFoundException();
+      }
+      await this.itemService.deleteRelation(orderToDelete.items)
```

Comment:
this could be replaced by cascade options I guess, did you check?

Reply:
I used {cascade: true}  but it didn't work. I found this solution